Robus/src/reception.c

```diff
@@ -198,11 +205,21 @@ void Recep_GetCollision(volatile uint8_t *data)
     }
     else
     {
-        if(data_count == COLLISION_DETECTION_NUMBER)
+        if (data_count == COLLISION_DETECTION_NUMBER)
         {
+            // collision detection end
             LuosHAL_SetRxState(false);
-            // switch to get header.
-            ctx.rx.callback = Recep_GetHeader;
+            LuosHAL_ResetTimeout(0);
+            if (ctx.tx.status == TX_NOK)
```

Comment:
You can use a ternary operator if it is not against you guidelines. I suggest
ctx.rx.callback = (ctx.tx.status == TX_NOK) ? Recep_CatchAck : Recep_GetHeader;